integration-libs/cds/src/components/recent-searches/recent-searches.service.ts

```diff
@@ -18,11 +18,12 @@ export class RecentSearchesService {
 
   public get recentSearches$(): Observable<string[]> {
     this.addRecentSearchesListener();
-    return this.recentSearchesSource.asObservable();
+    return this.recentSearchesSource;
   }
+
   checkAvailability() {
     return interval(150).pipe(
-      concatMap((_) => of((<any>this.winRef.nativeWindow).Y_TRACKING)),
+      concatMap((_) => of((<any>this.winRef.nativeWindow)?.Y_TRACKING)),
       take(5),
       takeWhile((result: any) => !result.recentSearches),
```

Comment:
One thing commented earlier, if this part `of((<any>this.winRef.nativeWindow).Y_TRACKING))` will return `undefined` i think it will also throw error on takeWhile when will try to access `recentSearches`, so probably you need also `result?.recentSearches` here

Reply:
ok, I can secure this element (`? lastSearches ) and as for the whole checkAvailability method, it doesn't work like "infinite check", but it only tries to get data from Y_TRACKING API once in max 5 intervals, and if this API doesn't exists, you should not call it endlessly.